_source/_posts/2022-06-01-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,604 @@
+  quantity Integer required min(0)
+  totalPrice BigDecimal required min(0)
+  status OrderItemStatus required
+}
+
+enum OrderItemStatus {
+  AVAILABLE, OUT_OF_STOCK, BACK_ORDER
+}
+
+relationship ManyToOne {
+  OrderItem{product(name) required} to Product
+}
+
+relationship OneToMany {
+  ProductOrder{orderItem} to OrderItem{order(code) required} ,
+  ProductCategory{product} to Product{productCategory(name)}
+}
+
+service Product, ProductCategory, ProductOrder, OrderItem with serviceClass
+paginate Product, ProductOrder, OrderItem with pagination
+microservice Product, ProductOrder, ProductCategory, OrderItem with product
+
+/**
+ * Entities for Invoice microservice
+ */
+// Invoice for sales
+entity Invoice {
+  code String required
+  date Instant required
+  details String
+  status InvoiceStatus required
+  paymentMethod PaymentMethod required
+  paymentDate Instant required
+  paymentAmount BigDecimal required
+}
+
+enum InvoiceStatus {
+  PAID, ISSUED, CANCELLED
+}
+
+entity Shipment {
+  trackingCode String
+  date Instant required
+  details String
+}
+
+enum PaymentMethod {
+  CREDIT_CARD, CASH_ON_DELIVERY, PAYPAL
+}
+
+relationship OneToMany {
+  Invoice{shipment} to Shipment{invoice(code) required}
+}
+
+service Invoice, Shipment with serviceClass
+paginate Invoice, Shipment with pagination
+microservice Invoice, Shipment with invoice
+
+/**
+ * Entities for notification microservice
+ */
+entity Notification {
+  date Instant required
+  details String
+  sentDate Instant required
+  format NotificationType required
+  userId Long required
+  productId Long required
+}
+
+enum NotificationType {
+  EMAIL, SMS, PARCEL
+}
+
+microservice Notification with notification
+```
+
+We define entities for each service and mark the entities as microservice entities. We also define relationships between entities, enums, and other options like pagination, service layer, etc. Please refer to JDL [Entities](https://www.jhipster.tech/jdl/entities-fields) and [relationships](https://www.jhipster.tech/jdl/relationships) documentation for more possibilities.
+
+Now, we are ready to run JHipster. Open a terminal window on the folder where you saved the JDL and run the below command.
+
+```bash
+jhipster jdl app.jdl --fork
+```
+
+This will create the applications with all their entities and specified configurations. You should be able to see the gateway application in action by running the below command on the **store** folder.
+
+```bash
+./gradlew # starts the spring boot application
+```
+
+You can find a [sample application on GitHub](https://github.com/oktadev/okta-java-spring-k8s-istio-microservices-example).
+
+## Create a GKE Cluster and Install Istio
+
+To deploy the stack to Google Kubernetes Engine, we need to create a cluster and install Istio. So let's begin by creating a cluster using Google Cloud SDK.
+
+### Create a cluster
+
+Ensure you are logged into the gcloud CLI and run the below command to create a GKE cluster.
```

Comment:
```suggestion
Ensure you are logged into the gcloud CLI and run the following command to create a GKE cluster.
```
see line 16 here: https://oktawiki.atlassian.net/wiki/spaces/DOC/pages/722929640/InfoDev+Style+Guide#2.-Terminology-quick-reference "Don't use to refer to a place in your doc. Use a cross reference to the heading, or use the following."